From 1ab1415a09bc98e02c9fbecbc279159ee909f328 Mon Sep 17 00:00:00 2001
From: Jon West <electrikjesus@gmail.com>
Date: Thu, 24 Oct 2019 13:10:59 -0400
Subject: [PATCH 6/8] Add old modprobe entry

Add legacy modprobe mode to installed boot options too

Conflicts:
	install/scripts/1-install

Add old modprobe method as a boot option (AUTO_LOAD=old)

add the needed string in grub.cfg
---
 boot/boot/grub/grub.cfg            | 1 +
 install/grub2/efi/boot/android.cfg | 1 +
 install/scripts/1-install          | 2 ++
 3 files changed, 4 insertions(+)

diff --git a/boot/boot/grub/grub.cfg b/boot/boot/grub/grub.cfg
index d140366..b1b0727 100644
--- a/boot/boot/grub/grub.cfg
+++ b/boot/boot/grub/grub.cfg
@@ -1,5 +1,6 @@
 set timeout=30
 set live=Live
 set debug_mode="Live DEBUG mode"
+set autoload_old="Legacy Modprobe mode"
 
 source /efi/boot/android.cfg
diff --git a/install/grub2/efi/boot/android.cfg b/install/grub2/efi/boot/android.cfg
index 8eb26cd..82c3458 100644
--- a/install/grub2/efi/boot/android.cfg
+++ b/install/grub2/efi/boot/android.cfg
@@ -85,6 +85,7 @@ export android bootefi grub kdir live src
 # Create main menu
 add_entry "$live" quiet
 add_entry "$debug_mode" DEBUG=2
+add_entry "$autoload_old" AUTO_LOAD=old
 if [ -s ($android)$kdir/install.img ]; then
 	add_entry "Installation" INSTALL=1
 fi
diff --git a/install/scripts/1-install b/install/scripts/1-install
index 0b54007..7d7873c 100644
--- a/install/scripts/1-install
+++ b/install/scripts/1-install
@@ -255,6 +255,7 @@ create_menulst()
 	echo -e "${GRUB_OPTIONS:-default=0\ntimeout=6\nsplashimage=/grub/android-x86.xpm.gz\n}root (hd0,$1)\n" > $menulst
 
 	create_entry "$OS_TITLE $VER" quiet $cmdline
+	create_entry "$OS_TITLE $VER (Legacy modprobe mode)" $cmdline androidboot.hardware=android_x86_64 AUTO_LOAD=old
 	create_entry "$OS_TITLE $VER (Debug mode)" $cmdline DEBUG=2
 	create_entry "$OS_TITLE $VER (Debug nomodeset)" nomodeset $cmdline DEBUG=2
 	create_entry "$OS_TITLE $VER (Debug video=LVDS-1:d)" video=LVDS-1:d $cmdline DEBUG=2
@@ -468,6 +469,7 @@ install_to()
 		[ -s efi/boot/grub/grubenv ] || ( printf %-1024s "# GRUB Environment Block%" | sed 's/k%/k\n/; s/   /###/g' > efi/boot/grub/grubenv )
 
 		echo -e 'set timeout=5\nset debug_mode="(DEBUG mode)"' > $grubcfg
+		echo -e 'set timeout=5\nset autoload_old="(Old Modprobe mode)"' > $grubcfg
 		# Our grub-efi doesn't support ntfs directly.
 		# Copy boot files to ESP so grub-efi could read them
 		if [ "$fs" = "fuseblk" ]; then
-- 
2.17.1

